scripts/umbrel-os/external-storage/monitor

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+# Load monitor-lib
+check_dependencies () {
+  for cmd in "$@"; do
+    if ! command -v $cmd >/dev/null 2>&1; then
+      echo "This script requires \"${cmd}\" to be installed"
+      exit 1
+    fi
+  done
+}
+
+check_dependencies readlink dirname
+UMBREL_ROOT="$(dirname $(readlink -f "${BASH_SOURCE[0]}"))/../../.."
+
+source "${UMBREL_ROOT}/scripts/monitor-lib"
+
+
+# Monitor external mount
+block_device="${1}"
+MOUNT_POINT="${2}"
+
+check_mount () {
+    # Check if Umbrel root is still bind mounted
+    if ! df -h "${UMBREL_ROOT}" | grep --quiet '/dev/sd'; then
+        echo "External device not bind mounted to Umbrel root"
+        return 1
+    fi
+
+    # Check if device is still mounted
+    if ! grep -q "${MOUNT_POINT}" /proc/mounts; then
+        echo "External device not mounted at ${MOUNT_POINT}"
+        return 1
+    fi
+
+    # Check if device is still registered
+    if ! cat /sys/block/${block_device}/ro; then
+        echo "External block device '${block_device}' not found"
+        return 1
+    fi
+
+    # Check if device is read-only
+    READ_ONLY=$(cat /sys/block/${block_device}/ro)
```

Comment:
This is checking the entire block device, I think we probably want to be checking the partition: e.g `/sys/block/sda/sda1/ro` instead of `/sys/block/sda/ro`.

Reply:
Ya can do as well to make more specific. I'll add that change and test when I start back testing